cli-plugins/socket/socket.go

```diff
@@ -35,16 +38,29 @@ func randomID() string {
 	return hex.EncodeToString(b)
 }
 
-func accept(listener *net.UnixListener, conn **net.UnixConn) {
-	go func() {
+// accept creates a new Unix socket connection
+// and sends it to the *net.UnixConn channel
+// it allows reconnects
+func accept(listener *net.UnixListener) <-chan *net.UnixConn {
+	connChan := make(chan *net.UnixConn, 1)
+
+	go func(connChan chan<- *net.UnixConn) {
 		for {
-			// ignore error here, if we failed to accept a connection,
-			// conn is nil and we fallback to previous behavior
-			*conn, _ = listener.AcceptUnix()
+			// this is a blocking call and will wait
+			// until a new connection is accepted
+			c, err := listener.AcceptUnix()
+
 			// perform any platform-specific actions on accept (e.g. unlink non-abstract sockets)
-			onAccept(*conn, listener)
+			onAccept(listener)
+			// retry accepting a connection if there was an error
+			if err != nil {
+				continue
```

Comment:
I'm not sure I follow. I kept the previous behavior in-tact here since we _always_ called `onAccept`, even on failure.

https://github.com/docker/cli/blob/master/cli-plugins/socket/socket.go#L43-L45

Reply:
moved this to only be called when the connection is nil. I don't know the history behind this, but it seems to remove the socket file, which I assume is done when the connection failed as a cleanup routine?